cs169/BJC-Teacher-Tracker-App
# Country field added for international schools

### [Pivotal Tracker Link](https://www.pivotaltracker.com/story/show/187216522)


[tracker]: https://www.pivotaltracker.com/story/show/your-story-id

## What this PR does:
This PR adds the country field to the school model. Whenever a new school is created, the option to select a country is available. Moreover, the country data is introduced in tables, teacher and school information panels and in the backend. The UI logic either renders a textfield if the country selected when creating a school is not the US or a dropdown of US states if the selected country is the US. 

### Include screenshots
<img width="1695" alt="Screenshot 2024-03-11 at 5 02 43 PM" src="https://github.com/cs169/BJC-Teacher-Tracker-App/assets/41014369/b43bf02f-d1f6-4dbb-a3b7-663f1d1e518a">
<img width="1046" alt="Screenshot 2024-03-11 at 5 03 08 PM" src="https://github.com/cs169/BJC-Teacher-Tracker-App/assets/41014369/8abe0f8d-0b0c-4153-acd3-57f80afdc742">
<img width="1720" alt="Screenshot 2024-03-11 at 5 03 15 PM" src="https://github.com/cs169/BJC-Teacher-Tracker-App/assets/41014369/4b3e4907-a599-4286-8cb2-2e0983f6c8fc">
<img width="1714" alt="Screenshot 2024-03-11 at 5 03 28 PM" src="https://github.com/cs169/BJC-Teacher-Tracker-App/assets/41014369/559cfa45-28f3-4a3b-9dcc-aeb69bc7d5b6">
<img width="1720" alt="Screenshot 2024-03-11 at 5 03 37 PM" src="https://github.com/cs169/BJC-Teacher-Tracker-App/assets/41014369/5154b291-575c-49f2-8b3c-1d34e20350fa">
<img width="1098" alt="Screenshot 2024-03-11 at 5 04 43 PM" src="https://github.com/cs169/BJC-Teacher-Tracker-App/assets/41014369/95bc20a2-6f93-450b-a58e-eedc55fc88b2">
<img width="1110" alt="Screenshot 2024-03-11 at 5 04 56 PM" src="https://github.com/cs169/BJC-Teacher-Tracker-App/assets/41014369/69b24dd7-198c-4a25-bd6d-c588b9c7ed94">

### Who authored this PR?
- @razztech overall implementation, testing, code refactoring and UI logic
- @perryzjc pair-programming help for debugging teacher requests table

### How should this PR be tested?
1. Create a new request as a teacher 
2. Input a school that is not present in the dropdown and click enter 
3. Select the country of your choice outside the US and type a random state in the State field
4.  Submit your request. 
5. Login as an admin and see that the country selected and the state are present in the data. 

### What do the specs/features test?
- The feature tests if there is a country outside of the US selected then we should be able to type in the name of the state for that country instead of selecting from the usual US states dropdown. 

### Are there edge cases to watch out for?
No.

### Migration
- 20240307225738_add_country_to_schools.rb = adds the country field to the school model;
- Should use `rails db:migrate` to update the model

### New Gems
- Installed the `country_select` gem for the country dropdown that includes the `countries` and `unaccent` gems
- Need to run `bundle install` to use these gems

### Checklist:
